irohad/ordering/on_demand_os_transport.hpp

```diff
@@ -23,32 +26,65 @@ namespace iroha {
     namespace transport {
 
       /**
-       * Type of round indexing
+       * Type of round indexing by blocks
+       */
+      using BlockRoundType = uint64_t;
+
+      /**
+       * Type of round indexing by reject before new block commit
        */
-      using RoundType = uint64_t;
+      using RejectRoundType = uint32_t;
+
+      /**
+       * Type of proposal round
+       */
+      using RoundType = std::pair<BlockRoundType, RejectRoundType>;
+
+      /**
+       * Class provides hash function for RoundType
+       */
+      class RoundTypeHasher {
+       public:
+        std::size_t operator()(const RoundType &val) const {
+          return boost::hash_value(val);
+        }
+      };
 
       /**
        * Notification interface of on demand ordering service.
        */
       class OdOsNotification {
        public:
+        /**
+         * Type of stored proposals
+         */
+        using ProposalType = std::shared_ptr<shared_model::interface::Proposal>;
+
+        /**
+         * Type of stored transactions
+         */
+        using TransactionType =
+            std::shared_ptr<shared_model::interface::Transaction>;
+
+        /**
+         * Type of inserted collections
+         */
+        using CollectionType = std::vector<TransactionType>;
+
         /**
          * Callback on receiving transactions
          * @param transactions - vector of passed transactions
          */
-        virtual void onTransactions(
-            std::vector<std::shared_ptr<shared_model::interface::Transaction>>
-                transactions) = 0;
+        virtual void onTransactions(const CollectionType &transactions) = 0;
```

Comment:
This interface suggests that `transactions` ownership is not passed to receiver, which does not seem to be the case, because they are effectively moved to a proposal. Maybe use rvalue reference explicitly instead of ref-to-const?